Framework/PythonInterface/mantid/plots/plotfunctions.py

```diff
@@ -324,15 +330,43 @@ def create_subplots(nplots, fig=None):
         fig.clf()
     # annoyling this repl
     nplots = nrows * ncols
-    gs = GridSpec(nrows, ncols)
     axes = np.empty(nplots, dtype=object)
-    ax0 = fig.add_subplot(gs[0, 0], projection=PROJECTION)
-    axes[0] = ax0
-    for i in range(1, nplots):
-        axes[i] = fig.add_subplot(gs[i // ncols, i % ncols], projection=PROJECTION)
-    axes = axes.reshape(nrows, ncols)
+    cbar_axis = None
+    fig.set_layout_engine(layout="tight")
+
+    if add_cbar_axis:
+        # The right most column of the GridSpec is made a SubGridSpec to facilitate the colour bar
+        # This is done so that the colour bar can have a close spacing to the right most column
+        # Keeping the colour bar in the GridSpec rather than alongside it means it behaves much nicer
```

Comment:
Could this be reworded slightly? It seems a bit subjective and not clear what behaviour it is actually doing.